add svg@1.5.0 w/ git auto-update

nolsherry opened the pull request:
PR for #9054 
@kennynaoh, @pvnr0082t, please help me review this PR, thanks! 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Profile of the lib
- Git repository (required): https://github.com/kbwood/svg
- Official website (optional, not the repository): http://keith-wood.name/svg.html
- NPM package url (optional):
- GitHub / Bitbucket popularity (required):
  - Count of stars: 148
  - Count of watchers: 19
  - Count of forks: 48
- NPM download stats (optional):
  - Downloads in the last day:
  - Downloads in the last week:
  - Downloads in the last month:
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [X] This lib was not found on cdnjs repo
- [X] No already exist / duplicated issue and PR
- [X] The lib has notable popularity
  - [X] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [ ] More than 500 downloads stats per month on npm registry
- [ ] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [X] Has valid tags for each versions (for git auto-update)
- [X] Auto-update setup
- [X] Auto-update target/source is valid.
- [X] Auto-update filemap is correct.
# Git commit checklist
- [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [X] The parent of the commit(s) in the PR is not old than 3 days.
- [X] Pull request is sending from a non-master branch with meaningful name.
- [X] Separate unrelated changes into different commits.
- [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [X] Close corresponding issue in commit message
- [X] Mention related issue(s), people in commit message, comment.

close #9054, cc @kbwood

pvnr0082t commented:
LGTM

PeterDaveHello commented:
Thanks for your contribution, by this PR, I just found a point of the web-minify-helper needed to be improved, and now improved, please update your web-minify-helper and try to minify again, thanks!

nolsherry commented:
@pvnr0082t, @kennynaoh, I have minified these files again, please help me review this PR, thanks!

kennynaoh commented:
@nolsherry 
Should we also need `jquery.svg.pack.js` and `jquery.svg.compat-1.0.1.js` in the old versions?
If not, please manually add old versions without these two file.
Maybe you can discuss with author.
Thanks!

PeterDaveHello commented:
What's going on here?

nolsherry commented:
@kennynaoh the author does not response me in [svg/issues#23](https://github.com/kbwood/svg/issues/23) for a while.
Should we still waiting for the response?

nolsherry commented:
@kennynaoh I have confirm with the author that we do not need `jquery.svg.pack.js` and `jquery.svg.compat-1.0.1.js` in the old versions. ([svg/issues#23](https://github.com/kbwood/svg/issues/23)) 
Please help me review this PR, thanks!

kennynaoh commented:
@nolsherry Please add old versions manually without those two files. I think you also need to change the directory name, like `Release_1_4_5` -> `1.4.5`, and update `.gitignore` to ignore those version with unusual prefix. Thanks!

PeterDaveHello commented:
Any updates here? Thanks!

nolsherry commented:
@PeterDaveHello sorry for the late reply. I have updated this PR.
@kennynaoh I have added old versions and changed the naming of the old version folders.
In additions, I have modified `.gitignore` file to  ignore those version with unusual prefix.
Please help me review this PR, thanks!

nolsherry commented:
@kennynaoh I have removed the pack file and compat file in the old versions. 
Please help me review this PR, thanks!

nolsherry commented:
@kennynaoh I have updated `package.json`  in the first commit, 
please help me review this PR, thank!

nolsherry commented:
@kennynaoh I have removed `jquery.svg.js.map` in `v1.0.0` and `v1.0.1`.
In addition, I have tried to use `minify.sh`  to generate `map` file for `jquery.svg.css`. 
Could you tell me how to do? Thanks!

sufuf3 commented:
Hi @nolsherry  Do you want to continue to work on this?
Thank you.

nolsherry commented:
@sufuf3, I am fixing this PR, thanks for your reminding! :)

nolsherry commented:
@pvnr0082t, Please help me review this PR, thanks!

sufuf3 commented on ajax/libs/kbw.svg/package.json:
```diff
@@ -0,0 +1,37 @@
+      {
+        "basePath": "",
+        "files": [
+          "jquery.svg.*"
```
I think we can use `jquery.svg*`, as the homepage said in page `Extension Plugins`.
What do you think?

Thank you.

nolsherry replied:
@sufuf3, thanks your suggestion. I think `jquery.svg*` is inappropriate, because it will fetch the files we do't need.  
For example, `jquery.svgdom.js`.

sufuf3 replied:
@nolsherry The homepage said we need them.
http://keith-wood.name/svg.html --> `Extension Plugins`

nolsherry commented:
@sufuf3, If the lib name in commit message as the same package.json's name field (kbw.svg)

The commit message first character should be capital, such as from add svg@1.5.0 ... to Add svg@1.5.0 ...

If the second detail commit message can mention what versions be added, it will be useful to us to maintain.

nolsherry commented:
@sufuf3, I have modified lib name in commit message as the same with package.json's name field (`kbw.svg`), and modified the commit message first character into capital.
In addition, I have mentioned what old versions be added in the second commit message, please help me review this PR, thanks!

sufuf3 commented:
Let's modify the second commit message because the average length of a commit message is 50/72 characters.

Form 
```
Add old versions of kbw.svg, cc #9054

Manually add old versions: 1.0.0, 1.0.1, 1.1.0, 1.1.1, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.3.2, 1.4.0, 1.4.1, 1.4.2, 1.4.3, 1.4.4, 1.4.5.
In addition, change versions' name from git repo
because some of old versions come with unusual prefix.
```
To
```
Add old versions of kbw.svg, cc #9054

Manually add old versions: 1.0.0, 1.0.1, 1.1.0, 1.1.1, 1.2.0, 1.2.1,
1.3.0, 1.3.1, 1.3.2, 1.4.0, 1.4.1, 1.4.2, 1.4.3, 1.4.4, 1.4.5.
In addition, change versions' name from git repo because some of old
versions come with unusual prefix.
```

maruilian11 commented on ajax/libs/kbw.svg/package.json:
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "kbw.svg",
+  "filename": "jquery.svg.min.js",
+  "description": "This plugin allows you to control/interact with SVG elements within your HTML documents. * Native SVG support in major browsers (but only IE9+). * Create a new SVG element or wrap an existing inline SVG element. * Animate SVG attributes with the svganim module. * Apply jQuery class/attribute/event functionality to SVG elements with the svgdom module. * Draw graphs with the svggr ...
```
I think it's better to not use "list of description". How about `A jQuery plugin that lets you interact with an SVG canvas.` from the offical website?
http://keith-wood.name/svg.html

maruilian11 commented on ajax/libs/kbw.svg/package.json:
```diff
@@ -0,0 +1,37 @@
+    "plot",
+    "ui"
+  ],
+  "version": "1.5.0",
```
Let's put `version` below `name`? Make it easy to check the version later.

sufuf3 commented:
@nolsherry Would you fix this PR, please?
Many thanks.

sufuf3 commented:
@nolsherry thanks for contribution.
Would you mind working on this and helping improve this PR?

Many thanks.

PeterDaveHello commented:
@sufuf3 you can make a new PR.

dtunes commented:
What else is needed for this PR to be approved?
